client/qiskit_serverless/core/files.py

```diff
@@ -138,7 +138,7 @@ def upload(
         with tracer.start_as_current_span("files.upload"):
             with open(file, "rb") as f:
                 with requests.post(
-                    os.path.join(self._files_url, "upload"),
+                    os.path.join(self._files_url, "upload/"),
```

Comment:
This change was due to use a `POST` without a finished slash: https://docs.djangoproject.com/en/dev/ref/settings/#std-setting-APPEND_SLASH

Honestly, I couldn't look at that configuration yet so when I review it we can decide if we should start adding the slash at the end in all our end-points or what.